### PR TITLE
Parse storage constraints from bundle data

### PR DESCRIPTION
**Pull request by axw**
Add support for storage in bundles, so we can deploy bundles of charms that use the storage feature.

### CODE REVIEW

**Comment by axw** [comment]
@frankban would you mind taking a look please? I'm not sure what the impact of changing the signature of Verify/VerifyWithCharms is, but I think it needs to be done all the same.

File: bundledata.go [hunk 1]
```diff
@@ -303,6 +318,11 @@ func (verifier *bundleDataVerifier) verifyServices() {
 		if err := verifier.verifyConstraints(svc.Constraints); err != nil {
 			verifier.addErrorf("invalid constraints %q in service %q: %v", svc.Constraints, name, err)
 		}
+		for storageName, storageConstraints := range svc.Storage {
```
Comment (frankban) [inline bundledata.go]: Is there any way on which storageName itself can be invalid?

**Comment by frankban** [comment]
Thanks for this branch!
Callers (basically juju-core and the charm store) will need to reflect the change on Verify(WithCharms), which is backward incompatible, but this is an unstable version anyway.
So :+1: with a question about validating storage keys.

File: bundledata.go (hunk 1 above)
Comment (axw) [inline bundledata.go]: Yes, there is, thanks for picking that up. I've added validation of the storage name too, along with a test.

**Comment by axw** [comment]
$$merge$$

**Pull request merged**